### ISSUE TITLE
Mismatch in parameters, help text and manuscript

### ISSUE THREAD

**Issue by schorlton** [phac-nml/ecoli_serotyping#92]
Hi @kbessonov1984,

Thanks for the great tool!

I noticed that there appears to be a mismatch between ECTyper defaults, help message descriptions and the ECTyper manuscript.

Specifically these lines: https://github.com/phac-nml/ecoli_serotyping/blob/1.0.0/ectyper/commandLineOptions.py#L71-L101

In the manuscript, I see:
>the hits are filtered using default thresholds of >=95 % identity and >=90 % coverage for O-antigen alleles and >=90 % identity and >=50 % coverage for H-antigen alleles

Therefore, I understand:

Parameter | Default | Help message | Manuscript
-- | -- | -- | --
-opid | 90 | 90 | 95
-hpid | 95 | 95 | 90
-opcov | 90 | 95 | 90
-hpcov | 50 | 50 | 50

</body>

</html>

Am I misunderstanding the parameters, missing other logic or should this be corrected? I am happy to contribute via PR once clarified.

Thanks!

**Comment by kbessonov1984** [comment]
Good day Sam,

Thank you for brining this to our attention, I fixed the help message values to match defaults via a special variable (pull latest commit `ed4996e` and install via `pip3 install .`). This is already addressed in version 2.0.0 of the tool released on a separate branch now undergoing active development (https://github.com/phac-nml/ecoli_serotyping/tree/v2.0.0). 

Due to historical reasons and additional practical testing in the field, the current "default" values are currently the best recommended ones (i.e. use the `Default` column values from your column for the best performance). During the development and testing stage there was a lot of tuning on internal and public E.coli datasets that resulted in oversights unfortunately. 

The O antigen typing is complex multi-stage process with various thresholds being applied including initial 95% identity threshold for list of potential candidate O antigens. The O antigen `-opid` value is currently sitting at 90% as we discovered that some O antigens have `wzm` and `wzt` genes with `%identity` values at 94%.

The reason why H antigen `-hpid` is lower than manuscript value sitting at 95% as we discovered that H antigens  are easier to type and public data showed no hits below 95% `%identity` so we increased value for robustness without affecting previous results.

In addition O antigen typing in ECTYPER is complex so thresholds specified by `-opid` `-opcov` do not always fully apply directly as one might expect. The main logic is described by [L38 onwards](https://github.com/phac-nml/ecoli_serotyping/blob/0545926d04654a46dea83ef814ba1dc1a2db7452/ectyper/predictionFunctions.py#L38) code. Briefly,  potential O antigens are being discovered using a low similarity thresholds of 95% identity and 48% coverage  specified by the `MIN_O_IDENTITY_LS` and `MIN_O_COVERAGE_LS` constants defined in [`definitions.py`](https://github.com/phac-nml/ecoli_serotyping/blob/master/ectyper/definitions.py). If only a single O antigen candidate is found, then 95% identity and 48% coverage thresholds are applied on O antigen allele hits instead of user specified values. If more than 1 candidate O antigens are found, then thresholds specified by the `-opid` and `-opcov` parameters are applied on the candidate alleles which currently default to 90% identity and 90% coverage. In the case of the H antigen filters only `-hpid` and `-hpcov` parameter values are used without any special logic which currently default to 90%  and 50%.
